client/src/components/content/CharacterShow.js

```diff
@@ -1,37 +1,75 @@
 import React, { useState, useEffect } from "react"
+import { useParams } from "react-router-dom";
+import VoteButtons from "./VoteButtons.js"
 
 const CharacterShow = (props) => {
   const [character, setCharacter] = useState({})
-  
+  const [totalVotes, setTotalVotes] = useState(0)
+  const [userVote, setUserVote] = useState(false)
+  const [clicked, setClicked] = useState(0)
+  const params = useParams()
+
   const getCharacter = async () => {
-    const characterId = props.match.params.id
+    const characterId = params.id
     try {
       const response = await fetch(`/api/v1/characters/${characterId}`)
       if (!response.ok) {
         const errorMessage = `${response.status} (${response.statusText})`
         const error = new Error(errorMessage)
-        throw(error)
+        throw (error)
       }
       const body = await response.json()
       setCharacter(body.character)
-    }catch(error) {
+      setTotalVotes(body.character.votes[0].sum)
+      setUserVote(body.character.voted)
+    } catch (error) {
       console.error("error in character show", error.Message)
     }
   }
-  
-  useEffect (()=> {
+
+ 
+
+  useEffect(() => {
     getCharacter()
-  }, [])
+  }, [clicked])
+
+  const addVote = async (event) => {
+    event.persist();
+    const newVote = {
+      voteValue: parseInt(event.currentTarget.value)
+    }
+    const characterId = params.id
+    try {
+      await fetch(`/api/v1/characters/${characterId}/vote`, {
```

Comment:
Proper HTTP RESTful pathing would be plural `votes` for this POST path
https://restfulapi.net/http-methods/